### PR TITLE
chore: Update dependencies

### PR DESCRIPTION
**Pull request by reneleonhardt**
### Features
* Update Rust dependencies
* Update Maven dependencies
* Update Docker images
* Let Dependabot update Rust, Maven and Docker versions
* Fix 5 security vulnerabilities in codegen swagger (and maybe more in docker images)

| Library | Vulnerability | Severity |
|-|-|-|
| com.google.guava:guava | CVE-2023-2976 | MEDIUM |
| | CVE-2020-8908 | LOW |
| org.json:json | CVE-2022-45688 | HIGH |
| | CVE-2023-5072 | |
| org.yaml:snakeyaml | CVE-2022-1471 | |

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
-image: Visual Studio 2017
+image: Visual Studio 2019
```
Comment (fussybeaver) [inline appveyor.yml:1]: I'm not 100% sure docker is working in this one: https://www.appveyor.com/docs/windows-images-software/#docker ...

Comment (reneleonhardt) [reply appveyor.yml:1]: Yeah, I'm at a loss how a PR can possibly pass that test 😅
What can I do, should I revert that file?

Comment (fussybeaver) [reply appveyor.yml]: Yeah, try to revert that file, it should go through

**Comment by fussybeaver** [comment]
LGTM

**Pull request merged**